pkg/api/v1/types.go

```diff
@@ -2292,6 +2292,12 @@ type Binding struct {
 	Target ObjectReference `json:"target"`
 }
 
+// Preconditions must be fulfilled before an operation (update, delete, etc.) is carried out.
+type Preconditions struct {
+	// Specifies the target UID.
+	UID *types.UID `json:"uid,omitempty"`
```

Comment:
> Is it supposed to check if this isn't a new object recreated after previous deletion?

Yes.

Alternatively we can use RV as a precondition, but that has false negative. For example, when a user tries to delete an object, but the object is updated by another party, the deletion fails.